pass.el

```diff
@@ -413,6 +476,103 @@ If SUBDIR is nil, return the entries of `(password-store-dir)'."
+
+(defun pass-view--reload-otp (buffer &optional last-token)
+  "Reload BUFFER's OTP token and countdown for URI, using LAST-TOKEN."
+  (when (buffer-live-p buffer)
+    (with-current-buffer buffer
+      (save-excursion
+        (goto-char (point-min))
+        (when (search-forward "otp: " nil t)
+          (let* ((buf-modified (buffer-modified-p))
+                 (remaining-secs (pass-view--otp-remaining-secs))
+                 (token (if (or (not last-token)
+                                (string= remaining-secs "30"))
+                            (password-store-otp-token (pass-view-entry-name buffer))
+                          last-token)))
+            (let ((lb (line-beginning-position))
+                  (le (search-forward "remaining"))
+                  (inhibit-read-only t))
+              (remove-text-properties lb le '(read-only nil))
+              (goto-char (line-beginning-position))
+              (re-search-forward "\\b[0-9]\\{6\\}\\b")
+              (replace-match token)
+              (re-search-forward "\\b[0-9]\\{2\\}s\\b")
+              (replace-match (concat remaining-secs "s"))
+              (add-text-properties lb le '(read-only t)))
+            (set-buffer-modified-p buf-modified)
+            (run-at-time 1 nil #'pass-view--reload-otp buffer token)))))))
+
+(defun pass-view--mask-otp ()
+  "Replace otpauth uri for an OTP token and remaining seconds."
+  (save-excursion
+    (goto-char (point-min))
+    (when (search-forward "otpauth://" nil t)
+      (setq pass-view--uri
+            (buffer-substring (line-beginning-position) (line-end-position)))
+      (let ((buf-modified (buffer-modified-p)))
+        (delete-region (line-beginning-position) (line-end-position))
+        (let ((ro-start (point)))
+          (insert (concat "otp: 000000 - 00s remaining"))
+          (mapcar (lambda (btn)
+                    (insert " ")
+                    (insert-button (car btn) 'action (cdr btn)))
+                  `(("[Copy]" . ,(lambda (_) (password-store-otp-token-copy (pass-view-entry-name))))
+                    ("[QR Code]" . ,(lambda (_) (pass-view--qrcode)))
+                    ("[Edit URI]" . ,(lambda (_) (pass-view--unmask-otp)))))
+          (add-text-properties ro-start (point) '(read-only t front-sticky t)))
+        (pass-view--reload-otp (current-buffer))
+        (set-buffer-modified-p buf-modified)))))
+
+(defun pass-view--unmask-otp ()
+  "Show the buffer's entry OTP URI editable."
+  (save-excursion
+    (goto-char (point-min))
+    (when (re-search-forward "otp: \\b[0-9]\\{6\\}\\b" nil t)
+      (let ((lb (line-beginning-position))
+            (le (line-end-position))
+            (inhibit-read-only t))
+        (remove-text-properties lb le '(read-only nil))
+        (delete-region lb le))
+      (insert pass-view--uri)
+      (setq-local buffer-undo-list '()))))
+
+(defun pass-view--prepare-otp ()
+  "Look current buffer for OTP values and prepare for it.
```

Comment:
This sentence is a bit confusing IMHO.

What about:

"Look for an OTP value in the current buffer.
If one is found, setup the buffer to use OTP features."